### PR TITLE
feat: add --version flag to root cmd

### LINKED ISSUE #93: Implement version command [CLI-39]
Should return with the version of the CLI that is being used.

### PR DESCRIPTION
**Pull request by vprasanth**
### Description

- Add's simple `--version` flag to root cmd. 
- Added version to user-agent 

Took the logic from the https://github.com/auth0/platform-interface project, thanks @grounded042! The build-info package defines a few variables we don't yet use in the app -- but might be useful to have available. 

Current format: 
```
$ auth0 --version
auth0 version v0.1.4 fa0ac1b-dirty
```

### References
- https://auth0team.atlassian.net/browse/CLI-39
- https://auth0.slack.com/archives/C01KGNSRS4C/p1614379142063000

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used, if not `master`

Close #93

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -23,9 +46,9 @@ build:
 
 # Build a beta version of stripe for all supported platforms
```
Comment (jfatta) [inline Makefile:47]: Stripe? 😅

File: internal/cli/cli.go [hunk 2]
```diff
@@ -156,7 +157,7 @@ func (c *cli) setup(ctx context.Context) error {
 	if t.AccessToken != "" {
 		m, err := management.New(t.Domain,
 			management.WithStaticToken(t.AccessToken),
-			management.WithUserAgent(userAgent))
+			management.WithUserAgent(fmt.Sprintf("%v/%v", userAgent, strings.TrimPrefix(buildinfo.Version, "v"))))
```
Comment (jfatta) [inline internal/cli/cli.go:160]: Nice 📊

File: Makefile [hunk 3]
```diff
@@ -1,5 +1,28 @@
...
+PKG := github.com/auth0/$(NAME)
+BUILDINFOPKG := $(PKG)/internal/build-info
+
+## setup variables for build-info
+BUILDUSER := $(shell whoami)
+BUILDTIME := $(shell date -u '+%Y-%m-%d %H:%M:%S')
+VERSION := $(shell git describe --abbrev=0)
+GITCOMMIT := $(shell git rev-parse --short HEAD)
+
+GITUNTRACKEDCHANGES := $(shell git status --porcelain --untracked-files=no)
```
Comment (jfatta) [inline Makefile:14]: Is this in use?

File: Makefile (hunk 1 above)
Comment (vprasanth) [reply Makefile:47]: LOL so I guess what I borrowed was also borrowed 🤣

File: Makefile (hunk 3 above)
Comment (vprasanth) [reply Makefile:14]: It's used in the if statement to append `-dirty` to the git commit if files in the tree were modified during build

Comment (jfatta) [reply Makefile:14]: Oh sorry I missed that (still not used to the gh mobile app)

**Review by jfatta — APPROVED** [review]
🪄

**Review by Widcket — APPROVED** [review]

**Pull request merged**